saga/README.md

```diff
@@ -0,0 +1,121 @@
+# Saga Pattern
+
+This example demonstrates how to implement the [Saga pattern](https://microservices.io/patterns/data/saga.html) for realizing distributed transactions across multiple microservices, in a safe and reliable way using change data capture.
+
+Similarly to the outbox pattern, this implementation avoids unsafe dual writes to a service's database and Apache Kafka by channeling all outgoing messages through the originating service's database and capturing them from there using CDC and Debezium.
+
+There are three services involved:
+
+* _order-service:_ originator and orchestrator of the Saga
+* _customer-service:_ approves or rejects the customer's credit needed to fulfill an incoming order
+* _payment-service_ executes the payment associated to an incoming order
+
+## Running the Example
+
+Build and start up:
+
+```console
+$ mvn clean verify
+```
+
+```console
+$ docker-compose up --build
+```
+
+Register the connectors for the different services:
+
+```console
+$ http PUT http://localhost:8083/connectors/order-outbox-connector/config < register-order-connector.json
+$ http PUT http://localhost:8083/connectors/payment-outbox-connector/config < register-payment-connector.json
+$ http PUT http://localhost:8083/connectors/credit-outbox-connector/config < register-credit-connector.json
+```
+
+Place an order:
+
+```console
+$ http POST http://localhost:8080/orders < requests/place-order.json
+```
+
+Examine the emitted messages for `payment` and `credit-approval` in Apache Kafka:
+
+```console
+$ docker run --tty --rm \
+    --network saga-network \
+    debezium/tooling:1.1 \
+    kafkacat -b kafka:9092 -C -o beginning -q \
+    -f "{\"key\":%k, \"headers\":\"%h\"}\n%s\n" \
+    -t payment.request
+
+$ docker run --tty --rm \
+    --network saga-network \
+    debezium/tooling:1.1 \
+    kafkacat -b kafka:9092 -C -o beginning -q \
+    -f "{\"key\":%k, \"headers\":\"%h\"}\n%s\n" \
+    -t credit-approval.request
+```
+
+Examine the saga state in the order service's database:
+
+```console
+$ docker run --tty --rm -i \
+        --network saga-network \
+        debezium/tooling:1.1 \
+        bash -c 'pgcli postgresql://orderuser:orderpw@order-db:5432/orderdb'
+
+select * from order.sagastate;
```

Comment:
this has to be table `purchaseorder.sagastate` right?

Reply:
Yes, indeed.